pkg/swiftring/funcs.go

```diff
@@ -85,46 +86,66 @@ func DeviceList(ctx context.Context, h *helper.Helper, instance *swiftv1beta1.Sw
 		}
 	}
 
-	// Get all OpenStackDataPlaneNodeSets that deploy the Swift service and
+	// Get all DataPlaneNodeSet inventories that deploy the Swift service and
 	// their used Swift disks
-	nodeSets := &dataplanev1.OpenStackDataPlaneNodeSetList{}
-	err = h.GetClient().List(context.Background(), nodeSets, listOpts...)
+	secrets := &corev1.SecretList{}
+	labelSelector := map[string]string{
+		"openstack.org/operator-name": "dataplane",
+		"inventory":                   "true",
+	}
+	labels := client.MatchingLabels(labelSelector)
+	listOpts = append(listOpts, labels)
+
+	err = h.GetClient().List(context.Background(), secrets, listOpts...)
 	if err != nil && !errors.IsNotFound(err) {
 		return "", "", err
 	}
-	for _, nodeSet := range nodeSets.Items {
-		for _, service := range nodeSet.Spec.Services {
-			if service == "swift" {
-				// Get the global disk vars first that are used for all
-				// nodes if not set otherwise per-node
-				var globalDisks []swiftv1beta1.SwiftDisk
-				if edpmSwiftDisks, found := nodeSet.Spec.NodeTemplate.Ansible.AnsibleVars[DataplaneDisks]; found {
-					err = json.Unmarshal(edpmSwiftDisks, &globalDisks)
-					if err != nil {
-						return "", "", err
-					}
+	for _, secret := range secrets.Items {
+		inventory, err := ansible.UnmarshalYAML(secret.Data["inventory"])
+		if err != nil {
+			return "", "", err
+		}
+		nodeSetGroup := inventory.Groups[secret.Labels["openstackdataplanenodeset"]]
+
+		if slices.Contains(nodeSetGroup.Vars["edpm_services"].([]string), "swift") {
```

Comment:
I had to change `[]string` to `[]interface {}` to make this work:

`if slices.Contains(nodeSetGroup.Vars["edpm_services"].([]interface {}), "swift") {
`